### ISSUE TITLE
Support for non-integer quantities

### ISSUE THREAD

**Issue by felicitus** [Sylius/Sylius#8078]
Some shops require non-integer quantities. For example, if you want to sell fabric, the shop usually allows the customer to specify a certain length, typically based on a base of meters. So a customer must be able to type in 0.1 to receive 0.1 meters.

I'm aware of that the base could be 10cm, in order to get one meter, the customer would order 10 10cm pieces, but this is a very ugly solution, causes confusion at the customer's side and increases workload at the shop.

TL;DR: Support non-decimal quantities.

**Comment by psihius** [comment]
@felicitus Sylius deals with integers for a reason - floating-point calculations have issues.
In any case, your specific case is a highly custom requirement that requires a lot of changes and probably is outside the scope of the project at this point. I'd recommend visiting the Slack server Sylius has - I know at least 1 person there who did such custom stuff with length and price calculations, maybe he/she will share some details with you.

**Comment by felicitus** [comment]
It's sad that Sylius went the easy way (integers) instead of the right way (bcmath) :(
